qa/tasks/cephfs/test_volumes.py

```diff
@@ -7653,6 +7701,818 @@ def test_subvolume_snapshot_clone_with_no_wait_not_enabled(self):
+        self.config_set('mgr', 'mgr/volumes/clone_delay_every_regfile', 2)
+        self.run_ceph_cmd(f'fs subvolume snapshot create {v} {sv} {ss}')
+        self._wait_till_rbytes_is_right(v, sv)
+
+        self.run_ceph_cmd(f'fs subvolume snapshot clone {v} {sv} {ss} {c}')
+        self._wait_for_clone_to_be_in_progress(c)
+
+        p = 0.0
+        while p < 1.0:
+            try:
+                o = self.get_ceph_cmd_stdout('status --format json-pretty')
+                o = json.loads(o)
+                pev = o['progress_events']
+            except KeyError as e:
+                try:
+                    self.__check_clone_state('completed', clone=c, max_count=1)
+                except:
+                    msg = ('Didn\'t find expected entries in dictionary '
+                           '"progress_events" which is obtained from the '
+                           'output of command "ceph status".\n'
+                           f'Exception - {e}\npev -\n{pev}')
+                    raise Exception(msg)
+
+            pev = self._rm_global_recovery_event(pev)
+
+            # XXX: it it is assumed that by this point cloning must have
+            # begun.
+            if len(pev) < 1:
+                    break
+            elif len(pev) > 1:
+                raise RuntimeError('For 1 clone "ceph status" output has 2 '
+                                   'progress bars, it should have only 1 '
+                                   'progress bar.\npev -\n {pev}\no -\n{o}')
+
+            # ensure that exactly 1 progress bar for cloning is present in
+            # "ceph status" output
+            msg = ('"progress_events" dict in "ceph status" output must have '
+                   f'exactly one entry.\nprogress_event dict -\n{pev}')
+            self.assertEqual(len(pev), 1, msg)
+
+            v = pev.popitem()[1] # get value from dict
+
+            # make sure tha the  progress bar is for clone initiated in this
+            # test.
+            self.assertIn(f'Subvolume "{c}" has been', v['message'])
+
+            # let's make sure that clone finishes. for this let's wait till
+            # value of variable "p" reaches 1.0. let's make sure that
+            # happens.
+            p = v['progress']
+
+            time.sleep(2)
+
+        self._wait_for_clone_to_complete(c)
+
+    def test_progress_bar_for_4_clones(self):
+        '''
+        X = number of clones issued.
+        Y = number of threads that can be spawned for async cloning.
+
+        Test that one progress bar is printed when X <= Y. Also, test that
+        this progress bar shows average of progresses made by all clones.
+        '''
+        v = self.volname
+        sv = 'sv1'
+        ss = 'ss1'
+        c = self._gen_subvol_clone_name(4)
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self._do_subvolume_io(sv, None, None, 3, 1024)
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_delay', 10)
```

Comment:
Since, the config values are retained across tests. So we should revert the configs to default at the end of the tests. We did encounter issue earlier due to this. Or we can have a method which reverts all the configs to default and it can be called at the end of the tests which changed the config values.

Reply:
I have created a separate PR fixing this: https://github.com/ceph/ceph/pull/57627